### PR TITLE
Tesla changes, and tesla supply packs.

### PR DESCRIPTION
**Pull request by XeonFaux**
Add's the tesla parts to cargonia's supply packs.
Changes t he tesla to generate power from the PA.
If the tesla reaches -100 energy, lose a ball, if all balls lost, tesla dies.

### CODE REVIEW

**Comment by trifornt-tinfornt** [comment]
How fast does energy drop? This wouldn't really matter if it goes down -1 per BYOND second or something.

**Comment by danhibiki** [comment]
Looks good, mind defining an attack_hand for energy ball while your at it? right now, it gibs the user because its inheriting it from singularity.

File: code/modules/power/singularity/particle_accelerator/particle.dm [hunk 1]
```diff
@@ -38,7 +38,8 @@
 	if (A)
 		if(ismob(A))
 			toxmob(A)
-		if((istype(A,/obj/machinery/the_singularitygen))||(istype(A,/obj/singularity/)))
+		if((istype(A,/obj/machinery/the_singularitygen))||(istype(A,/obj/singularity/))||(istype(A,/obj/singularity/energy_ball
```
Comment (danhibiki) [inline code/modules/power/singularity/particle_accelerator/particle.dm:41]: isn't this excessive? energy_ball is a child of singularity, which is already checked for by istype.

Comment (TDcoolguy300) [inline code/modules/power/singularity/particle_accelerator/particle.dm:41]: Technically yes

Comment (TDcoolguy300) [inline code/modules/power/singularity/particle_accelerator/particle.dm:41]: I retract my comment

**Comment by Nokington** [comment]
sure

**Pull request merged**